src/dbnode/storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -184,9 +197,13 @@ func (s *commitLogSource) ReadData(
 	}
 
 	var (
-		fsOpts         = s.opts.CommitLogOptions().FilesystemOptions()
-		filePathPrefix = fsOpts.FilePathPrefix()
+		// Emit bootstrapping gauge for duration of ReadData
+		doneBootstrapping      = s.metrics.data.emitBootstrapping
```

Comment:
Hm, did you mean this?:
```go
		doneBootstrapping      = s.metrics.data.emitBootstrapping()
```

Reply:
Re: why didn't this break, as a meta point if we wanted to be more "type safe" about it we could make it return an `io.Closer` than calls the method.

```go
import "github.com/m3db/m3x/close"

var _ close.Closer = closerFn(nil)

// Implement close.Closer
func (f closerFn) Close() {
	f()
}

// Now to call it

func myThing() close.Closer {
	return closerFn(func() { /* code */})
}
```

Reply:
Fixed the function call.  I get what you're saying, but seems like it really hurts readability for something that would have been caught by a test if it was really important (we don't currently test whether metrics are being emitted properly.) I think I'm just gonna make the fix and leave as is for now